realm/realm-library/src/main/cpp/io_realm_internal_Property.cpp

```diff
@@ -66,15 +69,17 @@ JNIEXPORT jlong JNICALL Java_io_realm_internal_Property_nativeCreatePersistedPro
 }
 
 JNIEXPORT jlong JNICALL Java_io_realm_internal_Property_nativeCreatePersistedLinkProperty(JNIEnv* env, jclass,
-                                                                                          jstring j_name_str,
+                                                                                          jstring j_internal_name,
+                                                                                          jstring j_public_name,
                                                                                           jint type,
                                                                                           jstring j_target_class_name)
 {
     try {
-        JStringAccessor name(env, j_name_str);
+        JStringAccessor public_name(env, j_public_name);
+        JStringAccessor internal_name(env, j_internal_name);
         JStringAccessor link_name(env, j_target_class_name);
         PropertyType p_type = static_cast<PropertyType>(static_cast<int>(type));
-        return reinterpret_cast<jlong>(new Property(name, p_type, link_name));
+        return reinterpret_cast<jlong>(new Property(internal_name, p_type, link_name, "", public_name));
```

Comment:
Are the parameters for the Property constructor, ok?:
`inline Property::Property(std::string name, PropertyType type, std::string object_type, std::string link_origin_property_name, std::string public_name)`
```suggestion
        return reinterpret_cast<jlong>(new Property(internal_name, p_type, "", link_name, public_name));
```

Reply:
Yes, they should be fine (at least I think so, and all tests pass). The only constructor that accepts the public_name is also used to support backlinks, so the number of arguments is a bit weird.